quixstreams/sources/community/file/origins/azure.py

```diff
@@ -0,0 +1,59 @@
+import os
+from io import BytesIO
+from pathlib import Path
+from typing import Generator
+
+from .base import ExternalOrigin
+
+try:
+    from azure.storage.blob import BlobServiceClient
+    from azure.storage.blob._container_client import ContainerClient
+except ImportError as exc:
+    raise ImportError(
+        f"Package {exc.name} is missing: "
+        'run "pip install quixstreams[azure]" to use AzureOrigin'
+    ) from exc
+
+__all__ = ("AzureOrigin",)
+
+
+class AzureOrigin(ExternalOrigin):
```

Comment:
The official name of this service is "Files". You can't go more generic than that 🤣, I think `AzureFilesOrigin` might be better in this case.
```suggestion
class AzureFilesOrigin(ExternalOrigin):
```